test/sinon-test.js

```diff
@@ -115,7 +115,18 @@ describe("sinon module", function () {
 
         describe("useFakeXMLHttpRequest", function () {
             it("should be the fakeXhr.useFakeXMLHttpRequest export from nise", function () {
-                assert.equals(sinon.useFakeXMLHttpRequest, fakeNise.fakeXhr.useFakeXMLHttpRequest);
+                // use full sinon for this test as it compares sinon instance
+                // proxyquire changes the instance, so `actual instanceof Sandbox` returns `false`
+                // see https://github.com/sinonjs/sinon/pull/1586#issuecomment-354457231
+                sinon = require("../lib/sinon");
+
+                var nise = require("nise");
+
+                sinon.spy(nise.fakeXhr, "useFakeXMLHttpRequest");
+                sinon.useFakeXMLHttpRequest();
+                assert.isTrue(nise.fakeXhr.useFakeXMLHttpRequest.called);
+
+                nise.fakeXhr.useFakeXMLHttpRequest.restore();
```

Comment:
This should be in an `afterEach`. If the assertion above fails, this line will never be called in its current form.

Reply:
Clean up moved to `afterEach`.